pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -254,11 +248,37 @@ contract LBPool is WeightedPool, Ownable2Step, BaseHooks {
         return IRouterCommon(router).getSender() == owner();
     }
 
+    /**
+     * @notice Checks if a weight change is ongoing before allowing liquidity removal.
+     * @dev If a weight change is ongoing, the function reverts with "removingLiquidityNotAllowed".
+     * @param router The address of the router.
+     */
+    function onBeforeRemoveLiquidity(
+        address router,
+        address,
+        RemoveLiquidityKind,
+        uint256,
+        uint256[] memory,
+        uint256[] memory,
+        bytes memory
+    ) public view override onlyVault returns (bool success) {
+        if (router != _trustedRouter) {
+            revert RouterNotTrusted();
+        }
+
```

Comment:
```suggestion
```

I don't think this is needed; you're not doing anything about the sender here.

Reply:
Agree - we don't know what their process is. It's possible the owner is an EOA or contract that mints the project tokens or something, that will send the BPT somewhere (e.g., to a multi-sig) for withdrawal, as that's when there is a lot of value and it matters.